spec/unit/provider/mount/mount_spec.rb

```diff
@@ -317,9 +317,16 @@
         @provider.mount_fs()
       end
 
-      it "should not mount the filesystem if it is mounted" do
-        allow(@current_resource).to receive(:mounted).and_return(true)
-        expect(@provider).not_to receive(:shell_out!)
+      it "should remount the filesystem if it is mounted and the options have changed" do
+        options = "rw,noexec,noauto"
+        @new_resource.options(%w{rw exec auto})
+        @provider.should_receive(:shell_out!).with("mount -t ext3 -o remount,rw,exec,auto /dev/sdz1 /tmp/foo")
```

Comment:
@JamesBelchamber 
This expectation is invalid here as @provider.mount_fs() never calls remount_fs. Its a def action_mount who calls action_remount where remount_fs gets invoked. 
Could you please move this test to  spec/unit/provider/mount_spec.rb ?